src/main/java/com/ibm/devops/connect/CloudPublisher.java

```diff
@@ -176,32 +181,32 @@ public static void uploadJobInfo(JSONObject jobJson) {
         System.out.println(url);
         System.out.println(jobJson.toString());
 
-        CloudPublisher.postToSyncAPI(url, payload.toString());
+        CloudPublisher.postToSyncAPI(url, payload.toString(), instanceNum);
     }
 
-    public static void uploadJobStatus(JSONObject jobStatus) {
-        String url = CloudPublisher.getSyncApiUrl() + JENKINS_JOB_STATUS_ENDPOINT_URL;
-        CloudPublisher.postToSyncAPI(url, jobStatus.toString());
+    public static void uploadJobStatus(JSONObject jobStatus, int instanceNum) {
+        String url = CloudPublisher.getSyncApiUrl(instanceNum) + JENKINS_JOB_STATUS_ENDPOINT_URL;
+        CloudPublisher.postToSyncAPI(url, jobStatus.toString(), instanceNum);
     }
 
-    public static String uploadBuild(String payload) throws Exception {
+    public static String uploadBuild(String payload, int instanceNum) throws Exception {
         CloudPublisher.ensureHttpClientInitialized();
-        String localLogPrefix= logPrefix + "uploadBuild ";
         String resStr = "";
-        String url = CloudPublisher.getBuildUploadUrl();
+        String url = CloudPublisher.getBuildUploadUrl(instanceNum);
         CloseableHttpResponse response = null;
+        List<Entry> entries = Jenkins.getInstance().getDescriptorByType(DevOpsGlobalConfiguration.class).getEntries();
```

Comment:
If we are doing this in all of these methods, I think we can just pass the entry to these functions.  The `instanceNum` doesn't seem to give any benefit other than looking the entry up again.